### PR TITLE
single register match

### PR DESCRIPTION
**Pull request by deeksha-developer**
image
image

### CODE REVIEW

File: src/app/userDashboard/registerMatches/page.tsx [hunk 1]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
-    const matchTime = new Date(dbTime);
-    const timeDifference = Number(matchTime) - Number(currentTime);
-    if (timeDifference <= 900000) {
-      setRoomPwd(true);
+  const getRegisterMatchWithId = async () => {
+    try {
+      const token: any = localStorage.getItem('jwtToken');
+      const regMAtch = await sendRequest(`room/rooms/${matchID}`, {
+        method: 'GET',
+        headers: { Authorization: `Bearer ${token}` },
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx:45]: we don't want this remove please mention in previous PR's as well

File: src/app/userDashboard/registerMatches/page.tsx [hunk 2]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
+      });
+      setMatchData(regMAtch.data.room);
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: this should only happen if status is 200 then not

File: src/app/userDashboard/registerMatches/page.tsx [hunk 3]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
+      if (regMAtch.status === 200) {
+        const formatDateTime = ` ${formatDate({ date: matchData?.dateAndTime })} and ${formatTime({ time: matchData?.dateAndTime, format: 'LT' })}`;
+        setMatchData({ ...regMAtch.data.room, dateAndTime: formatDateTime })
+       CountdownComponent(matchData?.dateAndTime, matchData?.roomUuid, setVisibleRooms)
+      }
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx:59]: else for this should be throw Error()

File: src/app/userDashboard/registerMatches/page.tsx [hunk 4]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
 
-  const getRoomidPwd = () => {
-    const currentTime = new Date();
-    const dbTime = `${matchData?.date}T${matchData?.time}`;
-    const matchTime = new Date(dbTime);
-    const timeDifference = Number(matchTime) - Number(currentTime);
-    if (timeDifference <= 900000) {
-      setRoomPwd(true);
+  const getRegisterMatchWithId = async () => {
+    try {
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx:41]: state for loader should be true and false in error and 200 absed on that show loader

File: src/app/userDashboard/registerMatches/page.tsx [hunk 5]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
+      const token: any = localStorage.getItem('jwtToken');
+      const regMAtch = await sendRequest(`room/rooms/${matchID}`, {
+        method: 'GET',
+        headers: { Authorization: `Bearer ${token}` },
+      });
+      setMatchData(regMAtch.data.room);
+      if (regMAtch.status === 200) {
+        const formatDateTime = ` ${formatDate({ date: matchData?.dateAndTime })} and ${formatTime({ time: matchData?.dateAndTime, format: 'LT' })}`;
+        setMatchData({ ...regMAtch.data.room, dateAndTime: formatDateTime })
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: res.MAtch should be regMatch type here and also optional chaining must be used

File: src/app/userDashboard/registerMatches/page.tsx [hunk 6]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
+       CountdownComponent(matchData?.dateAndTime, matchData?.roomUuid, setVisibleRooms)
+      }
+    } catch (err) {
+      toast.error("not show data ")
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: What does this error means please check what we are showing in user tournament screen
Something went wrong, please try again later!
Every message you write should be meaningful not just any sentece

File: src/app/userDashboard/registerMatches/page.tsx [hunk 7]
```diff
@@ -5,63 +5,54 @@ import { Navbar } from '../../../Components/Navbar/Navbar';
...
     getRegisterMatchWithId();
+    
   }, []);
 
-  useEffect(() => {
-    getRoomidPwd();
-  }, [matchData]);
 
   return (
     <div className={styles.main_container}>
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: If you have seen there uerTournament properly there is id given to this Div please check why and what is doing

File: src/app/userDashboard/registerMatches/page.tsx [hunk 8]
```diff
@@ -76,50 +67,49 @@ const regMatches = () => {
...
-                </span> */}
               </div>
               <div className={styles.squad_match}>
-                <span className={styles.register_match}>
+                <span className={styles.register_match_gamename}>
                   {matchData?.gameName}
                 </span>
                 <span className={styles.winning_prize}>
-                  Time : {matchData?.date} at {matchData?.time}
+                  Time: {matchData?.dateAndTime.toString()}
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: why we need to do toString()

File: src/app/userDashboard/registerMatches/page.tsx [hunk 9]
```diff
@@ -76,50 +67,49 @@ const regMatches = () => {
...
                 </span>
 
                 <div className={styles.winnings}>
                   <div>
                     <span className={styles.winning_prize}>WINNING PRIZE</span>
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: Have you checked in userTpurnament screen for same also if you understand react component this whole section is a reusable component

File: src/app/userDashboard/registerMatches/page.tsx [hunk 10]
```diff
@@ -76,50 +67,49 @@ const regMatches = () => {
...
+                          height={12}
+                        />
+                      </span>
+                      {matchData?.lastSurvival}
                     </span>
                   </div>
 
                   <div>
-                    <span className={styles.winning_prize}> Entry FEES</span>
+                    <span className={styles.winning_prize}> Entry FEES :</span>
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: check userTournament for words

File: src/app/userDashboard/registerMatches/page.tsx [hunk 11]
```diff
@@ -163,108 +153,23 @@ const regMatches = () => {
...
-                    </div>
-
-                  </div>
-                  <div  className={styles.review_close}>
-                  x
-                  </div>
-                </div>
-                
-              </div>
+              {/* <div key={index} className={`${styles.slide}`}> */}
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/registerMatches/page.tsx]: we donot need any comments

File: src/app/userDashboard/tournament/CountdownComponent.tsx [hunk 12]
```diff
@@ -1,19 +1,16 @@
 'use client';
-import React from 'react';
-import styles from '../../../styles/CountdownComponent.module.scss';
+const CountdownComponent = (dateAndTime: string, roomUuid: string, setVisibleRooms: React.Dispatch<React.SetStateAction<string[]>>) => {
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/CountdownComponent.tsx]: is this a component or just a function can't we understand a difference between a component and a function

File: src/app/userDashboard/tournament/CountdownComponent.tsx [hunk 13]
```diff
@@ -1,19 +1,16 @@
 'use client';
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/CountdownComponent.tsx]: is this is a page so why are we using use client

File: src/app/userDashboard/tournament/page.tsx [hunk 14]
```diff
@@ -61,16 +61,17 @@ function Tournament() {
   const [matchDetails , setMatchDetails] = useState<tournament>(initialValues)
   const router = useRouter();
   const regMatchRedirect = (matchID: string) => {
-
+console.log("matchID =>",matchID)
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this

File: src/app/userDashboard/registerMatches/page.tsx (hunk 11 above)
Comment (deeksha-developer) [reply src/app/userDashboard/registerMatches/page.tsx]: this comment code use  for your team member section

File: src/app/userDashboard/registerMatches/page.tsx (hunk 8 above)
Comment (deeksha-developer) [reply src/app/userDashboard/registerMatches/page.tsx]: It represent any object as a string

**Pull request merged**